File: app/src/main/java/com/example/pam_app/view/BucketListView.java
package com.example.pam_app.view;

import android.content.Context;

import com.example.pam_app.utils.listener.Clickable;
import com.example.pam_app.utils.listener.ClickableWithParameter;
import com.example.pam_app.model.Bucket;

import java.util.List;

public interface BucketListView {
    void bindSpendingBuckets(List<Bucket> model);

    void bindSavingsBuckets(List<Bucket> model);

    void launchBucketDetailActivity(int position);

    void launchAddBucketActivity();

    void collapseSpendingBuckets();

    void collapseSavingsBuckets();

    void bind(Context context, Clickable launchAddBucketActivity,
              ClickableWithParameter<Integer> launchBucketDetailActivity, final List<Bucket> bucketList);

    void setIsSpendingListEmpty(boolean isEmpty);

    void setIsSavingsListEmpty(boolean isEmpty);

    void onBucketAdded(final Bucket bucket);

    void drawSpendingBucketList();

    void drawSavingsBucketList();
}

File: app/src/main/java/com/example/pam_app/view/BucketListView.kt
package com.example.pam_app.view

import android.content.Context
import com.example.pam_app.model.Bucket
import com.example.pam_app.utils.listener.ClickableTarget
import kotlin.jvm.Synchronized

interface BucketListView {
    fun bindSpendingBuckets(model: List<Bucket?>?)
    fun bindSavingsBuckets(model: List<Bucket?>?)
    fun launchBucketDetailActivity(position: Int)
    fun launchAddBucketActivity()
    fun collapseSpendingBuckets()
    fun collapseSavingsBuckets()
    fun bind(context: Context, launchAddBucketActivity: () -> Unit,
             launchBucketDetailActivity: (Int) -> Unit, bucketList: List<Bucket?>?)

    fun setIsSpendingListEmpty(isEmpty: Boolean)
    fun setIsSavingsListEmpty(isEmpty: Boolean)
    fun onBucketAdded(bucket: Bucket?)
    fun drawSpendingBucketList()
    fun drawSavingsBucketList()
    fun onDeleteBucket(id: Int)
}